src/test/java/com/iexec/common/docker/client/AbstractDockerTests.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2022 IEXEC BLOCKCHAIN TECH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.iexec.common.docker.client;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.core.DefaultDockerClientConfig;
+import com.github.dockerjava.core.DockerClientConfig;
+import com.github.dockerjava.core.DockerClientImpl;
+import com.github.dockerjava.transport.DockerHttpClient;
+import com.github.dockerjava.zerodep.ZerodepDockerHttpClient;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.springframework.test.util.ReflectionTestUtils;
+
+abstract class AbstractDockerTests {
+
+    DockerClientInstance dockerClientInstance = new DockerClientInstance();
+    DockerClientInstance corruptClientInstance = getCorruptInstance();
+
+    DockerClientInstance getCorruptInstance() {
+        DockerClientConfig config = DefaultDockerClientConfig.createDefaultConfigBuilder()
+                .withDockerHost("tcp://localhost:11111")
+                .build();
+        DockerHttpClient httpClient = new ZerodepDockerHttpClient.Builder()
+                .dockerHost(config.getDockerHost())
+                .sslConfig(config.getSSLConfig())
+                .build();
+        DockerClient corruptedDockerClient = DockerClientImpl.getInstance(config, httpClient);
+        DockerClientInstance dockerClientInstance = new DockerClientInstance();
+        ReflectionTestUtils.setField(dockerClientInstance, "client", corruptedDockerClient);
```

Comment:
Wondering if a new constructor with a `DockerClient client` arg wouldn't be cleaner/useful

Reply:
If possible, I prefer not to modify the API during this rework.
I don't find `ReflectionTestUtils` to be wrong in implementation like the current one.
There are other places were it is being used and allows to get the job done.